source/transactions/tests/unified/mongos-unpin.yml

```diff
@@ -116,3 +116,17 @@ tests:
       - *assertPinnedServer
       - *insertOne
       - *assertNoPinnedServer
+
+  - description: unpin when a non-transaction read operation uses a session
+    operations:
+      - *startTransaction
+      - *insertOne
+      - name: commitTransaction
+        object: *session0
+      - *assertPinnedServer
+      - name: find
+        object: *collection0
+        arguments:
+          filter: { x: 1 }
```

Comment:
nit: This can be an empty document for simplicity.

Reply:
Just to see if it'd work, I tried manually modifying this PR by adding a blank line and I think the issue is gone now? I can revert the commit if you need to clean up the branch or anything - let me know.